cpp/src/parquet/column_reader.cc

```diff
@@ -980,6 +1006,12 @@ class ColumnReaderImplBase {
   // plain-encoded data.
   std::unordered_map<int, std::unique_ptr<DecoderType>> decoders_;
 
+  // Skip info for current page
+  PageSkipInfo* skip_info_{nullptr};
```

Comment:
If the page is v1 without page-index, what would this being?

Reply:
> If the page is v1 without page-index, what would this being?

In our internal system, this cannot happen. It may indeed happen in this place, but the way to deal with it is very simple, we only need to fallback to the original logical when invoking ``NextChunk``.

Reply:
IMO fallback or checking is important. Some cases might likely to read external table or disable column statistics in some wide table

Reply:
In current implementation, we only check it in ``GetColumnPageReader`` under debug mode. Fallback and more checks will be added later.